### PR TITLE
Adds minor improvements to readability

### PR DESCRIPTION
**Pull request by danielestevez**
Based on suggestions by @demobox in original PR https://github.com/jclouds/jclouds/pull/1213

### CODE REVIEW

File: providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java [hunk 1]
```diff
@@ -33,19 +33,20 @@
...
+      if (name == null || name.length() < minLength || name.length() > maxLength)
+         throw exception(name, "Can't be null or empty. Length must be " + minLength + " to " + maxLength + " symbols");
+      if (!CharMatcher.JAVA_LETTER_OR_DIGIT.matches(name.charAt(0)))
          throw exception(name, "Should start with letter/number");
 
       CharMatcher range = getAcceptableRange();
       if (!range.matchesAllOf(name))
-         throw exception(name, "Should have lowercase or uppercase ASCII letters, numbers, or dashes");
+         throw exception(name,
+               "Should have lowercase or uppercase ASCII letters, numbers, dashes, underscores and " + "periods");
```
Comment (nacx) [inline providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java]: No need to break the string, as it generates an unnecessary StringBuilder object.

File: providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java [hunk 2]
```diff
@@ -54,7 +55,7 @@ private CharMatcher getAcceptableRange() {
 
    protected IllegalArgumentException exception(String name, String reason) {
       return new IllegalArgumentException(
-            String.format("Object '%s' doesn't match Azure naming constraints. " + "Reason: %s.", name,
+            String.format("Object '%s' doesn't match Azure naming constraints. " + "Reason: %s", name,
```
Comment (nacx) [inline providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java]: Same here. There is no need to break the string.

Comment (danielestevez) [reply providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java]: :up: with no period

**Comment by demobox** [comment]
> Adds minor improvements to readability

Looks good to me - many thanks, Daniel! Is this for 2.1.x only, or also for master, incidentally?

**Comment by danielestevez** [comment]
I think so! I opened the PR against 2.1.x as in the previous one but @nacx will surely merge it to 2.1.x and master

**Comment by nacx** [comment]
Squashed and pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds/commit/39d770ff) and [2.1.x](http://git-wip-us.apache.org/repos/asf/jclouds/commit/ff95137f). Thanks @danielestevez!

**Pull request closed without merging**